<?xml version="1.0"?>
<robot name="/home/ayrtom/ROS2/dev/FQW_ws/src/surround_view_segbev/resource/descriptions/EgoVehicle.urdf" xmlns:xacro="http://ros.org/wiki/xacro">
  <webots>
    <plugin type="surround_view_segbev.plugins.EgoVehicleDriver.EgoVehicleDriver"/>
    <plugin type="webots_ros2_driver::Ros2IMU">
      <frameName>imu</frameName>
      <topicName>/ego_vehicle/imu</topicName>
      <inertialUnitName>imu</inertialUnitName>
      <accelerometerName>accelerometer</accelerometerName>
      <gyroName>gyroscope</gyroName>
    </plugin>
  </webots>
  <link name="base_link">
    <visual>
      <origin xyz="1.69 0 0.52" rpy="0 0 0"/>
      <geometry>
        <box size="6.6 2 1"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="1.69 0 0.52" rpy="0 0 0"/>
      <geometry>
        <box size="6.6 2 1"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0.67 0 1.64" rpy="0 0 0"/>
      <geometry>
        <box size="4.5 1.95 1.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0.67 0 1.64" rpy="0 0 0"/>
      <geometry>
        <box size="4.5 1.95 1.3"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="3.60031 0 1.09128" rpy="0 0.55 0"/>
      <geometry>
        <box size="2.5 1.88 1.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="3.60031 0 1.09128" rpy="0 0.55 0"/>
      <geometry>
        <box size="2.5 1.88 1.3"/>
      </geometry>
    </collision>
  </link>
  <link name="base_footprint">
  </link>
  <joint name="base_footprint_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_footprint"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="backwards_lights">
  </link>
  <joint name="base_link_backwards_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="backwards_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="rear_lights">
  </link>
  <joint name="base_link_rear_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="rear_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="brake_lights">
  </link>
  <joint name="base_link_brake_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="brake_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="antifog_lights">
  </link>
  <joint name="base_link_antifog_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="antifog_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="left_indicators">
  </link>
  <joint name="base_link_left_indicators_joint" type="fixed">
    <parent link="base_link"/>
    <child link="left_indicators"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="right_indicators">
  </link>
  <joint name="base_link_right_indicators_joint" type="fixed">
    <parent link="base_link"/>
    <child link="right_indicators"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="front_lights">
  </link>
  <joint name="base_link_front_lights_joint" type="fixed">
    <parent link="base_link"/>
    <child link="front_lights"/>
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796"/>
  </joint>
  <link name="gps">
  </link>
  <joint name="base_link_gps_joint" type="fixed">
    <parent link="base_link"/>
    <child link="gps"/>
    <origin xyz="-0.00613 0 2.3" rpy="0 0 0"/>
  </joint>
  <link name="camera_rear">
  </link>
  <joint name="base_link_camera_rear_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_rear"/>
    <origin xyz="-1.546999 0.007993 2.11955" rpy="-1.570796 0.785398 1.570704"/>
  </joint>
  <link name="camera_rear_depth">
  </link>
  <joint name="camera_rear_camera_rear_depth_joint" type="fixed">
    <parent link="camera_rear"/>
    <child link="camera_rear_depth"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="lidar_rear">
  </link>
  <joint name="base_link_lidar_rear_joint" type="fixed">
    <parent link="base_link"/>
    <child link="lidar_rear"/>
    <origin xyz="-1.67 0.000005 0.14" rpy="0 0 3.1415"/>
  </joint>
  <link name="camera_front_right">
  </link>
  <joint name="base_link_camera_front_right_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_front_right"/>
    <origin xyz="1.7379 -0.817 2.13828" rpy="-1.570795 0.785396 -3.140791"/>
  </joint>
  <link name="camera_front_right_depth">
  </link>
  <joint name="camera_front_right_camera_front_right_depth_joint" type="fixed">
    <parent link="camera_front_right"/>
    <child link="camera_front_right_depth"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="camera_front_blind">
  </link>
  <joint name="base_link_camera_front_blind_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_front_blind"/>
    <origin xyz="5.3622 0.01 0.732" rpy="0 0 0"/> <!-- rpy="-1.570796 0.695 -1.570796" -->
  </joint>
  <link name="camera_front_blind_depth">
  </link>
  <joint name="camera_front_blind_camera_front_blind_depth_joint" type="fixed">
    <parent link="camera_front_blind"/>
    <child link="camera_front_blind_depth"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="camera_front">
  </link>
  <joint name="base_link_camera_front_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_front"/>
    <origin xyz="4.12 0.01 1.64" rpy="-1.570796 -0.000005 -1.570796"/>
  </joint>
  <link name="camera_front_depth">
  </link>
  <joint name="camera_front_camera_front_depth_joint" type="fixed">
    <parent link="camera_front"/>
    <child link="camera_front_depth"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="camera_front_left">
  </link>
  <joint name="base_link_camera_front_left_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_front_left"/>
    <origin xyz="1.7379 0.836 2.13828" rpy="-1.570798 0.785396 -0.000802"/>
  </joint>
  <link name="camera_front_left_depth">
  </link>
  <joint name="camera_front_left_camera_front_left_depth_joint" type="fixed">
    <parent link="camera_front_left"/>
    <child link="camera_front_left_depth"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="lidar_front">
  </link>
  <joint name="base_link_lidar_front_joint" type="fixed">
    <parent link="base_link"/>
    <child link="lidar_front"/>
    <origin xyz="4.6 0 1.14" rpy="0 0 0"/>
  </joint>
  <link name="imu">
  </link>
  <joint name="base_link_imu_joint" type="fixed">
    <parent link="base_link"/>
    <child link="imu"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="gyroscope">
  </link>
  <joint name="base_link_gyroscope_joint" type="fixed">
    <parent link="base_link"/>
    <child link="gyroscope"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="compass">
  </link>
  <joint name="base_link_compass_joint" type="fixed">
    <parent link="base_link"/>
    <child link="compass"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="accelerometer">
  </link>
  <joint name="base_link_accelerometer_joint" type="fixed">
    <parent link="base_link"/>
    <child link="accelerometer"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <link name="engine_speaker">
  </link>
  <joint name="base_link_engine_speaker_joint" type="fixed">
    <parent link="base_link"/>
    <child link="engine_speaker"/>
    <origin xyz="4.325 0 0" rpy="0 0 0"/>
  </joint>
  <joint name="right_rear_sensor" type="continuous">
    <parent link="base_link"/>
    <child link="rear_right_wheel"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0 -0.85 0" rpy="0 0 0"/>
  </joint>
  <link name="rear_right_wheel">
    <visual>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </collision>
  </link>
  <joint name="rear_right_wheel_joint" type="fixed">
    <parent link="base_link"/>
    <child link="rear_right_wheel"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
  <joint name="left_rear_sensor" type="continuous">
    <parent link="base_link"/>
    <child link="rear_left_wheel"/>
    <axis xyz="0 -1 0"/>
    <origin xyz="0 0.85 0" rpy="0 0 3.14159"/>
  </joint>
  <link name="rear_left_wheel">
    <visual>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="1.570796 0 0"/>
      <geometry>
        <cylinder radius="0.4" length="0.3"/>
      </geometry>
    </collision>
  </link>
  <joint name="rear_left_wheel_joint" type="fixed">
    <parent link="base_link"/>
    <child link="rear_left_wheel"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
</robot>
